### ISSUE TITLE
提问：FCOS方法iou loss计算问题

### ISSUE THREAD

**Issue by ghost** [PaddlePaddle/PaddleDetection#6460]
### 问题确认 Search before asking

- [X] 我已经搜索过问题，但是没有找到解答。I have searched the question and found no related answer.

### 请提出你的问题 Please ask your question

代码：
        ious = (area_inter + 1.0) / (
            area_predict + area_target - area_inter + 1.0)

我可以理解：只有positive样本所在的像素才计算iou损失（negative位置置0），分母分子都加一，这样算出来negative处的iou=1，其对应的iou loss = 0.

但是我认为这样会对positive样本的loss计算造成一定偏差，尤其是对于较小的目标，例如0.2/0.5 = 0.4, (0.2+1)/(0.5+1) = 0.8. 因此我觉得还是全部像素计算iou，再乘以positive_mask比较合适。

请问我的理解是否正确？

**Comment by ghost** [comment]
为了避免除以0的情况，只需要在分母加一个极小项（1e-7等等）就可以了

**Comment by nemonameless** [comment]
可以自己打印area_inter  area_predict  area_target 看看大小分布。你说的极端情况基本不存在，因为area_target不为0的时候基本都是几十几百出现的，个人觉得加个1只是平衡下分子分母的数量级。
是按FCOS作者版本复现的，https://github.com/tianzhi0549/FCOS/blob/master/fcos_core/layers/iou_loss.py#L36

**Comment by ghost** [comment]
收到 ，谢谢
